server/model/properties.go

```diff
@@ -45,13 +52,76 @@ type PropDef struct {
 
 // GetValue resolves the value of a property if the passed value is an ID for an option,
 // otherwise returns the original value.
-func (pd PropDef) GetValue(v string) string {
-	// v may be an id to an option.
-	opt, ok := pd.Options[v]
-	if ok {
-		return opt.Value
+func (pd PropDef) GetValue(v interface{}) (string, error) {
+	switch pd.Type {
+	case "select":
+		// v is the id of an option
+		id, ok := v.(string)
+		if !ok {
+			return "", ErrInvalidPropertyValueType
+		}
+		opt, ok := pd.Options[id]
+		if !ok {
+			return "", ErrInvalidPropertyValue
+		}
+		return strings.ToUpper(opt.Value), nil
+
+	case "date":
+		// v is a JSON string
+		date, ok := v.(string)
+		if !ok {
+			return "", ErrInvalidPropertyValueType
+		}
+		return pd.ParseDate(date)
+
+	case "person":
+		// v is a userid
+		userid, ok := v.(string)
+		if !ok {
+			return "", ErrInvalidPropertyValueType
+		}
+		// TODO
+		return userid, nil
+
+	case "multiSelect":
+		// v is a slice of strings containing option ids
+		ms, ok := v.([]interface{})
+		if !ok {
+			return "", ErrInvalidPropertyValueType
+		}
+		var sb strings.Builder
+		prefix := ""
+		for _, optid := range ms {
+			id, ok := optid.(string)
+			if !ok {
+				return "", ErrInvalidPropertyValueType
+			}
+			opt, ok := pd.Options[id]
+			if !ok {
+				return "", ErrInvalidPropertyValue
+			}
+			sb.WriteString(prefix)
+			prefix = ", "
+			sb.WriteString(strings.ToUpper(opt.Value))
+		}
+		return sb.String(), nil
 	}
-	return v
+	return fmt.Sprintf("%v", v), nil
+}
+
+func (pd PropDef) ParseDate(s string) (string, error) {
+	// s is a JSON snippet of the form: {"from":1642161600000} in milliseconds UTC
+	// The UI does not yet support date ranges.
```

Comment:
The UI does support date ranges.

![Screen Shot 2022-01-12 at 2 49 35 PM](https://user-images.githubusercontent.com/12704875/149227006-08fb97f0-9afb-4f19-800a-800deebcd166.png)

You will need to support JSON - {"from":1641513600000,"to":1642161600000} for now.
If when we support time, the full object is - 
```
export type DateProperty = {
    from?: number
    to?: number
    includeTime?: boolean
    timeZone?: string
}
```

Reply:
Fixed via https://github.com/mattermost/focalboard/pull/2110/commits/7190bf108beba3a1e9cdf0c2984fc4304c49de34